### ISSUE TITLE
`editor::SelectPrevious` does nothing if the user creates a selection below

### ISSUE THREAD

**Issue by marcospb19** [zed-industries/zed#15390]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Open a buffer with repeated words in separate lines so you can navigate using `editor::SelectPrevious`.
2. Go to the middle of the file, use `SelectPrevious` once to land at 2 selections.
3. Create a selection below all of the previous by calling `editor::SelectNext` (can also be done with the mouse drag + alt).
4. If you succeeded at 2 and 3, then subsequent calls to `SelectPrevious` won't do anything.

This is weird, `SelectNext` works after `SelectPrevious`, but the opposite isn't true.

### Environment

Zed: v0.145.1 (Zed)
OS: Linux X11 arch unknown
Memory: 31.2 GiB
Architecture: x86_64

**Comment by marcospb19** [comment]
yup
